server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -101,6 +101,26 @@ protected ClusterState execute(ClusterState currentState, AddColumnRequest reque
 
         request = addMissingParentColumns(request, currentTable);
         HashMap<ColumnIdent, List<Reference>> tree = buildTree(request.references());
+
+        // Request can contain parent columns of the new columns in order to be able to build and traverse the tree.
```

Comment:
Just a heads up: This won't be safe if there are concurrent dynamic column updates.

Whoever is sending the request may not have the latest cluster state and may include columns which already exist.
You'd have to combine this with the exists checks before or also use ` Reference exists = currentTable.getReference(ref.column());` again.

Reply:
thx! updated code a bit, now OID assignment is done together with allExists. Also added OID existence checks - it helped to spot some bugs, one is still under investigation.